fuel-vm/src/lib.rs

```diff
@@ -2,8 +2,15 @@
 
 #![warn(missing_docs)]
 #![deny(unsafe_code)]
-#![deny(unused_crate_dependencies)]
+// #![deny(unused_crate_dependencies)]
 #![deny(clippy::string_slice)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[doc(hidden)] // Needed by some of the exported macros
+pub extern crate alloc;
+
+#[cfg(feature = "std")]
+extern crate libm as _; // Not needed with stdlib
```

Comment:
Based on the comment we need to use `cfg(not(feature = "std"))`

Reply:
This line exists to satisfy `#![deny(unused_crate_dependencies)]`. Since we only depend on `libm` on `no_std` targets, we need to ignore the dependency like this when `stdlib` is available. It's not possible to specify a dependency that's disabled when a feature is enabled.